internal/transport/transport_test.go

```diff
@@ -476,10 +483,42 @@ func setUpWithNoPingServer(t *testing.T, copts ConnectOptions, connCh chan net.C
 		}
 		framer := http2.NewFramer(conn, conn)
 		if err := framer.WriteSettings(); err != nil {
-			t.Errorf("Error at server-side while writing settings: %v", err)
+			t.Errorf("Error at server-side while writing settings: %v", framer.ErrorDetail())
 			close(connCh)
 			return
 		}
+		if respModeCh != nil {
+			if err := framer.WriteSettingsAck(); err != nil {
+				t.Errorf("Error while writing settings: %v", err)
+				return
+			}
+			go func() {
+				respondToPing := true
+				var lock sync.Mutex
+				for {
+					frame, err := framer.ReadFrame()
+					if err != nil {
+						return
+					}
+					select {
+					case data, ok := <-respModeCh:
+						if !ok {
+							return
+						}
+						respondToPing = data
```

Comment:
On another look, I think we can do without the `ok` check. Also since we have a non blocking read, do we want the channel to have a buffer?

Reply:
Reading on the empty channel leads to `default` in the `select` clause. The `ok` here is used to know when the channel is closed and all sent data has been received (empty **and** closed), and then to exit the goroutine (to avoid an infinite loop and a leaking goroutine). I guess `framer.ReadFrame()` could lead to exit, but I can't just leave this code keep reading a closed channel. Would you suggest a different approach?

Reply:
I agree and it makes sense. I'm okay with the correct approach.